runtime/doc/lua-plugin.txt

```diff
@@ -0,0 +1,433 @@
+*lua-plugin.txt*                     Nvim
+
+                            NVIM REFERENCE MANUAL
+
+		  Guide to developing Lua plugins for Nvim
+
+
+                                       Type |gO| to see the table of contents.
+
+==============================================================================
+Introduction                                                       *lua-plugin*
+
+This is a guide for getting started with Neovim plugin development. It is not
+intended as a set of rules, but as a collection of recommendations for good
+practices.
+
+For a guide to using Lua in Neovim, please refer to |lua-guide|.
+
+==============================================================================
+Type safety                                            *lua-plugin-type-safety*
+
+Lua, as a dynamically typed language, is great for configuration. It provides
+virtually immediate feedback.
+But for larger projects, this can be a double-edged sword, leaving your plugin
+susceptible to unexpected bugs at the wrong time.
+
+You can leverage LuaCATS https://luals.github.io/wiki/annotations/
+annotations, along with lua-language-server https://luals.github.io/ to catch
+potential bugs in your CI before your plugin's users do.
+
+------------------------------------------------------------------------------
+Tools						 *lua-plugin-type-safety-tools*
+
+- lua-typecheck-action https://github.com/marketplace/actions/lua-typecheck-action
+- llscheck https://github.com/jeffzi/llscheck
+- luacheck https://github.com/mpeterv/luacheck for additional linting
+- neodev.nvim https://github.com/folke/neodev.nvim
```

Comment:
Given that neovim now contains most type annotations is there still a reason for neodev? (Other than the luals configuration bits, which we could point out in the luals reference above)

Reply:
It can still be useful for getting lua-ls diagnostics for plugin dependencies.

I personally prefer to generate a .luarc.json, but that uses Nix and would probably be too niche/overwhelming to recommend in nvim docs.